Essentials/src/com/earth2me/essentials/I18n.java

```diff
@@ -157,10 +157,43 @@ public InputStream getResourceAsStream(final String string) {
             if (file.exists()) {
                 try {
                     return new FileInputStream(file);
-                } catch (FileNotFoundException ignored) {
-                }
+                } catch (FileNotFoundException ignored) {}
             }
             return null;
         }
     }
+
+    /**
+     * Reads .properties files as UTF-8 instead of ISO-8859-1, which is the default on Java 8/below.
+     * Java 9 fixes this by defaulting to UTF-8 for .properties files.
+     */
+    private static class UTF8PropertiesControl extends ResourceBundle.Control {
+        public ResourceBundle newBundle(String baseName, Locale locale, String format, ClassLoader loader, boolean reload) throws IOException {
+            String bundleName = toBundleName(baseName, locale);
+            String resourceName = toResourceName(bundleName, "properties");
+            ResourceBundle bundle = null;
+            InputStream stream = null;
+            if (reload) {
+                URL url = loader.getResource(resourceName);
+                if (url != null) {
+                    URLConnection connection = url.openConnection();
+                    if (connection != null) {
```

Comment:
Pretty sure URL#openConnection never returns null and just throws an IOException

Reply:
If the URL doesn't point to anything, `#openConnection()` will return null, so if someone puts an invalid/custom locale in then it'll cause a NPE here.